dirty_cat/super_vectorizer.py

```diff
@@ -287,6 +313,15 @@ def _clone_transformers(self):
         Note: typos are not detected here, they are left in and are detected
         down the line in `ColumnTransformer.fit_transform`.
         """
+        if isinstance(self.binary_cat_transformer, sklearn.base.TransformerMixin):
+            self.binary_cat_transformer_ = clone(self.binary_cat_transformer)
+        elif self.binary_cat_transformer is None:
+            self.binary_cat_transformer_ = OneHotEncoder(drop="if_binary")
+        elif self.binary_cat_transformer == "remainder":
+            self.binary_cat_transformer_ = self.remainder
+        else:
+            self.binary_cat_transformer_ = self.binary_cat_transformer
+
```

Comment:
I don't understand why we don't just add "drop='if_binary'" to the OneHotEncoder a couple lines below. It would lead to much simpler code.